src/dash-table/components/ControlledTable/index.tsx

```diff
@@ -858,10 +886,52 @@ export default class ControlledTable extends PureComponent<ControlledTableProps>
                     </div>))}
                 </div>
             </div>
-            {!this.displayPagination ? null : (
-                <div>
-                    <button className='previous-page' onClick={this.loadPrevious}>Previous</button>
-                    <button className='next-page' onClick={this.loadNext}>Next</button>
+            {!this.displayPagination || lastPage === 0 ? null : (
```

Comment:
I think the page navigation is getting complex enough to warrant its own component at this point.